### ISSUE TITLE
Return back default writer to sys.stdout

### ISSUE THREAD

**Issue by vanclmil** [udapi/udapi-python#57]
The [BaseWriter.before_process_document](https://github.com/udapi/udapi-python/blob/master/udapi/core/basewriter.py) changes `sys.stdout` to new IO component.
In the end, `BaseWriter.after_process_document `should return the default IO component to `sys.stdout` back again and close the new one.

My use case is using udapi within jupyter notebook. After using `Document.store_conlu` all printed output from jupyter goes to the file.

**Comment by martinpopel** [comment]
Thanks for reporting this bug. I hope I have fixed it now.
I admit redirecting `sys.stdout` is a kind of hack, but it makes implementing new writers much easier for the users
(in [Treex](https://github.com/ufal/treex) all `print` commands in writers had to use a special filehandle, which was easy to forget, which resulted in bugs when redirecting the output to files).

**Comment by vanclmil** [comment]
I'm afraid that the issue is not fixed yet.
Regarding this observation:
![jupyter_stdout](https://user-images.githubusercontent.com/32512717/77694106-cd22fd80-6fa9-11ea-8ccf-0b8419d8db94.PNG)
Jupyter stdout doesn't support `fileno()` method, which you are using now, so it crashes on this [line](https://github.com/udapi/udapi-python/blob/19b4e680b2ff7bbbec6060b9b97bcb1949f3196d/udapi/core/basewriter.py#L67).
Next thing is, that setting the `sys.stdout` to `sys.__stdout__` will not work either, because it does not reference the default `ipykernel` stdout.

I hotfixed the issue just using a member variable of BaseWriter to backup the default stdout and finally I set it back in `BaseWriter.after_process_document`
